pkgs/by-name/ve/vesktop/fix_read_only_settings.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/src/main/settings.ts b/src/main/settings.ts
+index 6fad97f..dfc64e3 100644
+--- a/src/main/settings.ts
++++ b/src/main/settings.ts
+@@ -26,8 +26,10 @@ function loadSettings<T extends object = any>(file: string, name: string) {
+ 
+     const store = new SettingsStore(settings);
+     store.addGlobalChangeListener(o => {
+-        mkdirSync(dirname(file), { recursive: true });
+-        writeFileSync(file, JSON.stringify(o, null, 4));
++        try {
++            mkdirSync(dirname(file), { recursive: true });
++            writeFileSync(file, JSON.stringify(o, null, 4));
++        } catch {}
```

Comment:
I don't *normally* like try-catches that silently fail, but let's be real here - no end user will ever see logs anyway

Reply:
IMO we should still log this in cases where the user did not intend to have a read-only `settings.json`.

```suggestion
+        } catch (e) {
+            console.error("Nix: Failed to write settings.", e);
+        }
```